src/record/record.js

```diff
@@ -125,34 +113,31 @@ Record.prototype.set = function( pathOrData, data ) {
 		return this;
 	}
 
-	if( arguments.length === 2 && utils.deepEquals( this._getPath( pathOrData ).getValue(), data ) ) {
-		return this;
-	}
-	else if( arguments.length === 1 && utils.deepEquals( this._$data, pathOrData ) ) {
+	var path = arguments.length === 1 ? undefined : pathOrData;
+	data = path ? data : pathOrData;
+
+	if ( utils.deepEquals( data, jsonPath.get( this._$data, path ) ) ) {
```

Comment:
This check can become a duplicate if there is a listener for ALL_EVENT. Not sure how to resolve that in a clean fashion.